integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/text_embedder.py

```diff
@@ -38,6 +39,7 @@ def __init__(
         api_url: Optional[str] = None,
         prefix: str = "",
         suffix: str = "",
+        truncate: Optional[TruncateMode] = None,
```

Comment:
Let's make this a `str|TruncateMode` union and convert the former to the latter.